src/snowflake/snowpark/column.py

```diff
@@ -303,6 +303,9 @@ def __bool__(self) -> bool:
             "'~' for 'not' if you're building DataFrame filter expressions. For example, use df.filter((col1 > 1) & (col2 > 2)) instead of df.filter(col1 > 1 and col2 > 2)."
         )
 
+    def __iter__(self) -> None:
+        raise TypeError("Column is not iterable")
```

Comment:
How about adding more text like if you use `sum`, `count`, please make sure you use `snowflake.snowpark.functions.sum` and `count` instead of the `sum`/`count` of python or other libraries (numpy also has sum...).